mlrun/utils/helpers.py

```diff
@@ -474,6 +474,98 @@ def normalize_name(name: str):
+
+    :param function_name: The original function name (can be None or empty string)
+
+    :return: A tuple of (modified_name, was_renamed, suffix) where:
+        - modified_name: The function name with the batch suffix (if not already present),
+          or empty string if input was empty
+        - was_renamed: True if the suffix was added, False if it was already present or if name was empty
+        - suffix: The suffix value that was used (or would have been used)
+
+    """
+    suffix = mlrun_constants.RESERVED_BATCH_JOB_SUFFIX
+
+    # Handle None or empty string
+    if not function_name:
+        return function_name, False, suffix
+
+    if not function_name.endswith(suffix):
+        return (
+            f"{function_name}{suffix}",
+            True,
+            suffix,
+        )
+    return function_name, False, suffix
+
+
+def strip_batch_job_suffix(
+    function_name: typing.Optional[str],
+) -> tuple[str, bool, str]:
+    """
+    Remove the batch job suffix if present.
+
+    This helper is used when converting a batch job back to a local/serving runtime for execution.
+    The launcher uses this to strip the `-batch` suffix when creating temporary local functions
+    from batch jobs for local execution.
+
+    :param function_name: The function name (may have -batch suffix, can be None or empty string)
+
+    :return: A tuple of (stripped_name, was_stripped, suffix) where:
+        - stripped_name: The function name without the batch suffix (if it was present),
+          or the original name if suffix wasn't present, or empty string/None if input was empty
+        - was_stripped: True if the suffix was removed, False if it wasn't present or if name was empty
+        - suffix: The suffix value that was removed (or would have been removed)
+
+    """
+    suffix = mlrun_constants.RESERVED_BATCH_JOB_SUFFIX
+
+    # Handle None or empty string
+    if not function_name:
+        return function_name, False, suffix
+
+    if function_name.endswith(suffix):
+        return function_name.removesuffix(suffix), True, suffix
+
+    return function_name, False, suffix
+
+
+def validate_function_name_for_batch_suffix(name: str, kind: str) -> None:
+    """Validate that serving and local function names don't end with reserved batch suffix.
+
+    This validation prevents users from manually creating functions with the reserved "-batch" suffix,
+    which is automatically appended by to_job() methods. Allowing manual use of this suffix would
+    defeat the purpose of automatic collision prevention.
+
+    :param name: Function name to validate
+    :param kind: Function kind ("serving", "serving_v2", "local", etc.)
+
+    :raises MLRunValueError: If the function name ends with the reserved batch suffix
+    """
+    import mlrun.errors
+
+    # Only validate for kinds that have to_job() methods that add the suffix
+    if kind not in ["serving", "serving_v2", "local"]:
```

Comment:
https://github.com/mlrun/mlrun/pull/8728#discussion_r2454528436

I suggest using the kinds defined in `RuntimeKinds`.
Please note that `"serving_v2"` is not there - can the kind have this value?

If not, please update also below `function_type` -> `kind`.

Reply:
serving_v2 is a subkind, you are correct. Will remove
Also will refactor